Allow tests to run when targeting i686 on a host where uname -m returns x86_64.

--- a/test/common.inc.orig	2024-11-07 22:51:15.382530584 +0100
+++ b/test/common.inc	2024-11-07 22:52:49.415441555 +0100
@@ -15,7 +15,9 @@ canonical_name() {
   esac
 }
 
-if [ -z "$MACHINE" ]; then
+if [ -n "$XBPS_TARGET_MACHINE" ]; then
+  MACHINE=$(canonical_name "${XBPS_TARGET_MACHINE%-musl}")
+else
   MACHINE=$(canonical_name $(uname -m))
 fi
 
